odc/stac/_dcload.py

```diff
@@ -2,13 +2,56 @@
 from typing import Any, Callable, Dict, Iterable, Optional, Sequence, Union
 from warnings import warn
 
+import datacube.utils.geometry
+import shapely.geometry
 import xarray as xr
 from datacube import Datacube
 from datacube.api.core import output_geobox
 from datacube.model import Dataset
 from datacube.utils.geometry import GeoBox
 
 
+def _geojson_to_shapely(xx: Any) -> shapely.geometry.base.BaseGeometry:
+    _type = xx.get("type", None)
+
+    if _type is None:
+        raise ValueError("Not a valid GeoJSON")
+
+    _type = _type.lower()
+    if _type == "featurecollection":
+        features = xx.get("features", [])
+        if len(features) == 1:
+            return shapely.geometry.shape(features[0]["geometry"])
+
+        return shapely.geometry.GeometryCollection(
+            [shapely.geometry.shape(feature["geometry"]) for feature in features]
+        )
+
+    if _type == "feature":
+        return shapely.geometry.shape(xx["geometry"])
+
+    return shapely.geometry.shape(xx)
+
+
```

Comment:
Would https://shapely.readthedocs.io/en/stable/manual.html#shapely.geometry.mapping help here?

Reply:
mapping seems to go the other way? Anything with `__geo_interface__` to GeoJSON. But, yes, I agree seems odd that there is no easy way to turn arbitrary GeoJSON into shapely geometry object. `shapely.geometry.shape` operates on geometry component of GeoJSON only and does not understand feature/featurecollection.

```python
assert shapely.geometry.mapping(gdf) == gdf.__geo_interface__
```